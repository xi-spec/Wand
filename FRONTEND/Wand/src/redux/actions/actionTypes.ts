export default {
  LOAD_PRODUCT_TYPE: 'LOAD_PRODUCT_TYPE',
  LOAD_ALL_PRODUCTS: 'LOAD_ALL_PRODUCTS',
  LOAD_NEW_PRODUCTS: 'LOAD_NEW_PRODUCTS',
  LOAD_PRODUCTS_LIST: 'LOAD_PRODUCTS_LIST',
  LOAD_PRODUCT_DETAIL: 'LOAD_PRODUCT_DETAIL',
  ADD_PRODUCT_TO_CART: ' ADD_PRODUCT_TO_CART',
  INCREASE_QUANTITY_CART: 'INCREASE_QUANTITY_CART',
  DECREASE_QUANTITY_CART: 'DECREASE_QUANTITY_CART',
  DELETE_PRODUCT_CART: 'DELETE_PRODUCT_CART',
  DELETE_ALL_CART: 'DELETE_ALL_CART',
  UPDATE_STOCK_API: 'UPDATE_STOCK_API',
  SEARCH_PRODUCT: 'SEARCH_PRODUCT',
  ADD_PRODUCT_TO_FAVOURITE: 'ADD_PRODUCT_TO_FAVOURITE',
  REGISTER: 'REGISTER',
  REGISTER_ERROR: 'REGISTER_ERROR',
  LOGIN: 'LOGIN',
  LOGIN_ERROR: 'LOGIN_ERROR',
  CLEAN_ISEXIST_STATUS: 'CLEAN_STATUS',
  CLEAN_LOGIN_ERROR: 'CLEAN_LOGIN_ERROR',
  CLEAN_REGISTED: 'CLEAN_REGISTED',
  REMOVE_FAVOURITE: 'REMOVE_FAVOURITE',
  LOGOUT: 'LOGOUT',
  UPDATE_USER: 'UPDATE_USER',
  PAYMENT_SUCCSSFULY: 'PAYMENT_SUCCESSFULY',
  PDATE_STOCK_API: 'PDATE_STOCK_API'
}
;
