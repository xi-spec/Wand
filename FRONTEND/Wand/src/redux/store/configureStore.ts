import reduxImmutableStatInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import rootReducer from '../reducers';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(reduxImmutableStatInvariant(), thunk))
  );
}

export default configureStore();
