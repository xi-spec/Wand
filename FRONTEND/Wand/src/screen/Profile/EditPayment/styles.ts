import { StyleSheet } from 'react-native';
import { COLORS, SIZE, FONTS } from '../../../constants/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 20,
    alignItems: 'center'
  },
  input: {
    width: 350,
    height: 40,
    marginBottom: 40,
    marginTop: 10,
    borderRadius: 5
  },
  error: {
    position: 'absolute',
    color: 'red',
    marginLeft: 10,
    top: 70
  },
  applyContainer: {
    position: 'absolute',
    width: '100%',
    height: 30,
    bottom: 20,
    backgroundColor: 'white'
  },
  apply: {
    width: '100%',
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.primary,
    borderRadius: 5
  },
  applyText: {
    color: 'white',
    fontFamily: FONTS.logoRegular,
    fontSize: SIZE.primary

  }
});

export default styles;
