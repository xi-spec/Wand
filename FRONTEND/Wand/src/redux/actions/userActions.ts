import axios from 'axios';
import actionTypes from './actionTypes';
import SERVER from '../../constants/server';
import { User, Product, ShoppingCart } from '../../models/initiaState';
import { Dispatch } from 'redux';

export function addProductToFavourite (product:Product, user:User, favouriteState:boolean) {
  let newUser;
  favouriteState
    ? newUser = { ...user, favourite: [...user.favourite, product] }
    : newUser = { ...user, favourite: user.favourite.filter(elem => elem._id !== product._id) };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, newUser);
    dispatch({
      type: actionTypes.ADD_PRODUCT_TO_FAVOURITE,
      data
    });
  };
}

export function register (email:string, name:string, password:string) {
  const query = { email, name, password };
  let actionObj;
  return async (dispatch:Dispatch) => {
    try {
      const { data, status } = await axios.post(SERVER.URL_REGISTER, query);
      actionObj = {
        type: actionTypes.REGISTER,
        data,
        status
      };
    } catch (error) {
      actionObj = {
        type: actionTypes.REGISTER_ERROR
      };
    }
    dispatch(
      actionObj
    );
  };
}

export function login (email:string, password:string) {
  const query = { email, password };

  return async (dispatch:Dispatch) => {
    let actionObj;
    try {
      const { data } = await axios.post(SERVER.URL_LOGIN, query);
      actionObj = {
        type: actionTypes.LOGIN,
        data

      };
    } catch (error) {
      actionObj = {
        type: actionTypes.LOGIN_ERROR
      };
    }

    dispatch(actionObj);
  };
}

export function logout (email:string, password:string) {
  const query = { email, password };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.post(SERVER.URL_LOGOUT, query);

    dispatch({
      type: actionTypes.LOGOUT,
      data
    });
  };
}

export function cleanIsExistStatus () {
  return {
    type: actionTypes.CLEAN_ISEXIST_STATUS
  };
}

export function cleanLoginError () {
  return {
    type: actionTypes.CLEAN_LOGIN_ERROR
  };
}

export function cleanRegisted () {
  return {
    type: actionTypes.CLEAN_REGISTED
  };
}

export function removeFavourite (product:Product, user:User) {
  const newUser = { ...user, favourite: user.favourite.filter(elem => elem._id !== product._id) };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, newUser);
    dispatch({
      type: actionTypes.REMOVE_FAVOURITE,
      data
    });
  };
}

export function updateAddress (street:string,
  city:string, country:string, postcode:string, user:User) {
  const query = { street, city, country, postcode };
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { address: query });
    dispatch({
      type: actionTypes.UPDATE_USER,
      data
    });
  };
}

export function updateCard (cardNumber:string, securtyCode:string, cardholder:string, expiryDate:string, user:User) {
  const query = { cardNumber, securtyCode, cardholder, expiryDate };
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { card: query });
    dispatch({
      type: actionTypes.UPDATE_USER,
      data
    });
  };
}

export function paymentSuccessfuly (user:User, cart:ShoppingCart) {
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { orderHistory: [...user.orderHistory, cart] });
    dispatch({
      type: actionTypes.PAYMENT_SUCCSSFULY,
      data
    });
  };
}
