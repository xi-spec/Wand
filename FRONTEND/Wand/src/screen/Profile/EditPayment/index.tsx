import React, { useState } from 'react';
import { Text, View, TouchableOpacity, ScrollView } from 'react-native';
import { User } from '../../../models/initiaState';
import styles from './styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { updateCard } from '../../../redux/actions/userActions';
import { Input } from 'react-native-elements';
import * as Animatable from 'react-native-animatable';

function EditPayment ({ user, navigation, actions }:{user:User, navigation:{goBack:Function}, actions:any}) {
  const { cardNumber, securtyCode, cardholder, expiryDate } = user.card || { undefined };

  const [userCardNumber, setUserCardNumber] = useState(cardNumber);
  const [userSecurtyCode, setUserSecurtyCode] = useState(securtyCode);
  const [userCardholder, setCardHoler] = useState(cardholder);
  const [userExpiryDate, setExpiryDate] = useState(expiryDate);
  const [error, setError] = useState(false);

  function apply () {
    userCardNumber.length !== 16 ? setError(true) : setError(false);

    if (userCardNumber.length === 16) {
      return (actions.updateCard(userCardNumber, userSecurtyCode, userCardholder, userExpiryDate, user),
      navigation.goBack()
      );
    }
  }

  return (
    <ScrollView
    keyboardShouldPersistTaps='handled'
    contentContainerStyle={styles.container}>
         <View>

         <Input
         label="Card number"
         containerStyle={styles.input}
         value={userCardNumber}
         onChangeText={(text) => setUserCardNumber(text)}
         placeholder='Card number'
         keyboardType='numeric'>
         </Input>
         {error && <Animatable.Text style={styles.error} animation='fadeInLeft' duration={500}> Please enter a valid card numbers </Animatable.Text>}

         <Input
         label="Security code"
        containerStyle={styles.input}
        value={userSecurtyCode}
        secureTextEntry
        maxLength={3}
        onChangeText={(text) => setUserSecurtyCode(text)}
         placeholder='security code'
         keyboardType='numeric'>
         </Input>

         <Input
         label="Cardholder name"
        containerStyle={styles.input}
        value={userCardholder}
        onChangeText={(text) => setCardHoler(text)}
         placeholder='cardholder name'>
         </Input>

         <Input
         label="Expiry date "
         containerStyle={styles.input}
        value={userExpiryDate}
        onChangeText={(text) => setExpiryDate(text)}
         placeholder='dd/mm/yy'>
         </Input>
         </View>

         <View style={styles.applyContainer}>
          <TouchableOpacity
          testID='apply'
          onPress={() => apply()}
          disabled={!userCardNumber || !userSecurtyCode || !userCardholder || !userExpiryDate}
           style={styles.apply}>
            <Text style={styles.applyText}>APPLY</Text>
          </TouchableOpacity>
        </View>

     </ScrollView>
  );
}

function mapStateToProps ({ user }) {
  return {
    user
  };
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      updateCard
    }, dispatch)
  };
}
export default connect(mapStateToProps, mapDispatchToProps)(EditPayment);
