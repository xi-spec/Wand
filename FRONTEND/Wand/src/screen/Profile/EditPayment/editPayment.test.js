import React from 'react';
import { render, fireEvent, act } from '@testing-library/react-native';
import configureStore from 'redux-mock-store';
import * as actions from '../../../redux/actions/userActions';
import { Provider } from 'react-redux';
import EditPayment from './index';

describe('Given a EditPayment component', () => {
  const mockStore = configureStore();
  let store;
  let navigation;

  beforeEach(() => {
    jest.spyOn(actions, 'updateCard').mockReturnValueOnce({ type: '' });
    store = mockStore({
      user: {
        card: {
        }
      }
    });
    navigation = {
      goBack: jest.fn()
    };
  });
  describe('When is render', () => {
    test('Then is mathced to snapshot', () => {
      const tree = render(<Provider store={store}><EditPayment navigation={navigation} /></Provider>);

      expect(tree).toMatchSnapshot();
    });

    test('Then is mathced to snapshot when hasnt user.card', () => {
      store = mockStore({
        user: {

        }
      });
      const tree = render(<Provider store={store}><EditPayment navigation={navigation} /></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When is typing in input', () => {
    test('Then is input value is change', () => {
      const { getByPlaceholderText } = render(<Provider store={store}><EditPayment navigation={navigation} /></Provider>);

      const cardNumberInput = getByPlaceholderText('Card number');
      const codeInput = getByPlaceholderText('security code');
      const cardHolderInput = getByPlaceholderText('cardholder name');
      const dateInput = getByPlaceholderText('dd/mm/yy');
      const text = ('hola');

      act(() => {
        fireEvent.changeText(cardNumberInput, text);
        fireEvent.changeText(codeInput, text);
        fireEvent.changeText(cardHolderInput, text);
        fireEvent.changeText(dateInput, text);
      });
      expect(cardNumberInput.props.value).toBe(text);
      expect(codeInput.props.value).toBe(text);
      expect(cardHolderInput.props.value).toBe(text);
      expect(dateInput.props.value).toBe(text);
    });
  });

  describe('When apply button is pressed', () => {
    test('Then actions.updateCard invoked', () => {
      store = mockStore({
        user: {
          card: {
            cardNumber: '1234567890123456',
            securtyCode: 'hello',
            cardholder: 'hello',
            expiryDate: 'hello'
          }
        }
      });

      const { getByTestId } = render(<Provider store={store}><EditPayment navigation={navigation}/></Provider>);

      fireEvent.press(getByTestId('apply'));
      expect(actions.updateCard).toHaveBeenCalled();
    });
    test('Then is rendered correct when cardNumber is incorrect', () => {
      store = mockStore({
        user: {
          card: {
            cardNumber: '123456789',
            securtyCode: 'hello',
            cardholder: 'hello',
            expiryDate: 'hello'
          }
        }
      });

      const tree = render(<Provider store={store}><EditPayment navigation={navigation}/></Provider>);

      fireEvent.press(tree.getByTestId('apply'));
      expect(tree).toMatchSnapshot();
    });
  });
});
