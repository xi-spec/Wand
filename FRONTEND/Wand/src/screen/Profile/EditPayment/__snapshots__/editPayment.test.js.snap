// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a EditPayment component When apply button is pressed Then is rendered correct when cardNumber is incorrect 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "margin": 20,
    }
  }
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Card number
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Card number"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value="123456789"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <Text
        forwardedRef={[Function]}
        style={
          Object {
            "color": "red",
            "marginLeft": 10,
            "opacity": 0,
            "position": "absolute",
            "top": 70,
            "transform": Array [
              Object {
                "translateX": -100,
              },
            ],
          }
        }
      >
         Please enter a valid card numbers 
      </Text>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Security code
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="numeric"
            maxLength={3}
            onChangeText={[Function]}
            placeholder="security code"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value="hello"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cardholder name
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="cardholder name"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value="hello"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Expiry date 
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="dd/mm/yy"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
            value="hello"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 20,
          "height": 30,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="apply"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          APPLY
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Given a EditPayment component When is render Then is mathced to snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "margin": 20,
    }
  }
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Card number
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Card number"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Security code
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="numeric"
            maxLength={3}
            onChangeText={[Function]}
            placeholder="security code"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cardholder name
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="cardholder name"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Expiry date 
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="dd/mm/yy"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 20,
          "height": 30,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="apply"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          APPLY
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Given a EditPayment component When is render Then is mathced to snapshot when hasnt user.card 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "margin": 20,
    }
  }
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Card number
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Card number"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Security code
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="numeric"
            maxLength={3}
            onChangeText={[Function]}
            placeholder="security code"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Cardholder name
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="cardholder name"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderRadius": 5,
            "height": 40,
            "marginBottom": 40,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": 350,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#86939e",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Expiry date 
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="dd/mm/yy"
            placeholderTextColor="#86939e"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "#242424",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "margin": 5,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 20,
          "height": 30,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2E3192",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="apply"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          APPLY
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
